backend/typescript/services/interfaces/userService.ts

```diff
@@ -5,47 +5,49 @@ interface IUserService {
    * Get user associated with id
    * @param id user's id
    * @returns a UserDTO with user's information
+   * @throws Error if user retrieval fails
    */
-  getUserById(id: number): UserDTO;
+  getUserById(userId: string): Promise<UserDTO>;
 
   /**
    * Get user associated with email
    * @param email user's email
    * @returns a UserDTO with user's information
+   * @throws Error if user retrieval fails
    */
-  getUserByEmail(email: string): UserDTO;
+  getUserByEmail(email: string): Promise<UserDTO>;
 
   /**
    * Get all user information (possibly paginated in the future)
    * @returns array of UserDTOs
+   * @throws Error if user retrieval fails
    */
-  getUsers(): Array<UserDTO>;
+  getUsers(): Promise<Array<UserDTO>>;
 
   /**
    * Create a user, email verification configurable
    * @param user the user to be created
    * @returns a UserDTO with the created user's information
+   * @throws Error if user creation fails
    */
-  createUser(user: CreateUserDTO): UserDTO;
+  createUser(user: CreateUserDTO): Promise<UserDTO>;
 
   /**
    * Update a user.
    * Note: the password cannot be updated using this method, use IAuthService.resetPassword instead
    * @param userId user's id
    * @param user the user to be updated
    * @returns a UserDTO with the updated user's information
+   * @throws Error if user update fails
    */
-  updateUserById(userId: number, user: UpdateUserDTO): UserDTO;
+  updateUserById(userId: string, user: UpdateUserDTO): Promise<UserDTO>;
 
   /**
    * Delete a user by id
    * @param userId user's userId
+   * @throws Error if user deletion fails
    */
-  deleteUserById(userId: number): void;
-
-  /**
-   * Delete a user by email
-   * @param email user's email
-   */
-  deleteUserByEmail(email: string): void;
```

Comment:
Deleted this since I originally thought there was a blocker for doing rollbacks, but on second thought I think it's a non-issue, will add it back.